internal/bminventory/inventory.go

```diff
@@ -292,8 +292,17 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 	return pullSecret, nil
 }
 
-func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context, params installer.V2RegisterClusterParams) (installer.V2RegisterClusterParams, error) {
+func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context, params installer.V2RegisterClusterParams, id strfmt.UUID) (installer.V2RegisterClusterParams, error) {
 	log := logutil.FromContext(ctx, b.log)
+
+	if params.NewClusterParams.APIVip != "" && len(params.NewClusterParams.APIVips) == 0 {
+		params.NewClusterParams.APIVips = []*models.APIVip{{IP: models.IP(params.NewClusterParams.APIVip), ClusterID: id}}
+	}
+
+	if params.NewClusterParams.IngressVip != "" && len(params.NewClusterParams.IngressVip) == 0 {
+		params.NewClusterParams.IngressVips = []*models.IngressVip{{IP: models.IP(params.NewClusterParams.IngressVip), ClusterID: id}}
+	}
+
```

Comment:
I have a feeling that you do not need to put `ClusterID: id` here explicitly because _something somewhere_ handles it for you. If you look just at the code below, you see

```
		params.NewClusterParams.ClusterNetworks = []*models.ClusterNetwork{
			{Cidr: models.Subnet(b.Config.DefaultClusterNetworkCidr), HostPrefix: b.Config.DefaultClusterNetworkHostPrefix},
		}
```

which in the DB also holds Cluster ID, but our code does not add it explicitly here